content/cluster-autoscaling/cluster-autoscaling.md

```diff
@@ -0,0 +1,225 @@
+
+There are other less obvious benefits to overprovisioning. One of the side effects of a highly utilized cluster is that pods will make less optimal scheduling decisions using the `preferredDuringSchedulingIgnoredDuringExecution` rule of Pod or Node Affinity. A common use case for this is to separate pods for a highly available application across availability zones using AntiAffinity. Overprovisioning can significantly increase the chance that a node of the correct zone is available.
+
+The amount of overprovisioned capacity is a careful business decision for your organization. At its core, it’s a tradeoff between performance and cost. One way to make this decision is to determine your average scale up frequency and divide it by the amount of time it takes to scale up a new node. For example, if on average you require a new node every 30 seconds and EC2 takes 30 seconds to provision a new node, a single node of overprovisioning will ensure that there’s always an extra node available, reducing scheduling latency by 30 seconds at the cost of a single additional EC2 Instance. To improve zonal scheduling decisions, overprovision a number of nodes equal to the number of availability zones in your EC2 Auto Scaling Group to ensure that the scheduler can select the best zone for incoming pods.
+
+### Prevent Scale Down Eviction
+
+Some workloads are expensive to evict. Unlike application serving, big data analysis, machine learning tasks, and test runners will eventually complete, but must be restarted if interrupted. The Cluster Autoscaler will attempt to scale down any node under the scale-down-utilization-threshold, which will interrupt any remaining pods on the node. This can be prevented by ensuring that pods that are expensive to evict are protected by a label recognized by the Cluster Autoscaler.
+
+Ensure that:
+
+* Protected workloads are be labeled with `cluster-autoscaler.kubernetes.io/safe-to-evict=false`
+
+## Advanced Use Cases
+
+### EBS Volumes
+
+Persistent storage is critical for building stateful applications, such as database or distributed caches. [EBS Volumes](https://aws.amazon.com/premiumsupport/knowledge-center/eks-persistent-storage/) enable this use case on Kubernetes, but are limited to a specific zone. These applications can be highly available if sharded across multiple AZs using a separate EBS Volume for each AZ. The Cluster Autoscaler can then balance the scaling of the EC2 Autoscaling Groups.
+
+Ensure that:
+
+* Node group balancing is enabled by setting `balance-similar-node-groups=true`.
+* Node Groups are configured with identical settings except for different availability zones and EBS Volumes.
+
+### Co-Scheduling
```

Comment:
I think we need to make some changes to this title. you can check my comments in quip doc.

Reply:
@ellistarn